### ISSUE TITLE
WinSxS msvcr90.dll conflict with Windows 10 Microsoft Feature update (Build 1803)

### ISSUE THREAD

**Issue by vis-123** [Nuitka/Nuitka#204]
It appears a Microsoft Windows 10 update is breaking Nuitka at compile time for me, as it is unable to find the msvcr90.dll in the expected WinSxS folder. The issue ONLY appears after the MS update; seems the update is moving or deleting the msvcr90.dll from the WinSxS folders that nuitka looks for its dependency. I've tested it with rollbacks on multiple devices, and can confirm the microsoft update is definitely the culprit.

Edit: After more research, the issue may be related to dll dependency caching, will try nuitka with cl arg ```--force-dll-dependency-cache-update``` to force an update.

The specific Windows10 affected version and build are:
```
Version: 1803
OS Build: 17134.471
```

My nuitka version info:
```
0.6.0.6
Python: 2.7.15 (v2.7.15:ca079a3ea3, Apr 30 2018, 16:30:26) [MSC v.1500 64 bit (AMD64)]
Executable: c:\python27\python.exe
OS: Windows
Arch: x86_64
```

Err output:
```
Traceback (most recent call last):
  File "C:\Python27\lib\site-packages\nuitka\__main__.py", line 180, in <module>
    main()
  File "C:\Python27\lib\site-packages\nuitka\__main__.py", line 174, in main
    MainControl.main()
  File "C:\Python27\lib\site-packages\nuitka\MainControl.py", line 844, in main
    standalone_entry_points = standalone_entry_points
  File "C:\Python27\lib\site-packages\nuitka\freezer\Standalone.py", line 1187, in copyUsedDLLs
    target_path
  File "C:\Python27\lib\shutil.py", line 96, in copyfile
    with open(src, 'rb') as fsrc:
IOError: [Errno 2] No such file or directory: 'c:\\windows\\winsxs\\amd64_microsoft.vc90.crt_1fc8b3b9a1e18e3b_9.0.30729.9247_none_08e394a1a83e212f\\msvcr90.dll'
```

**Comment by kayhayen** [comment]
Can you checkout the cache options of Nuitka on Windows and or erase it in AppData. I suspect that `depends.exe` output points to a DLL that n longer exists.

I guess that ought to make Nuitka update the cache for that item automatically, or warn, or something, but it can become outdated easily with any system update.

If you are willing to wait and preserve the "Nuitka" folder in AppData, I can make the change and use you to test it, would that be OK for you?

Yours,
Kay

**Comment by vi-s** [comment]
Hi @kayhayen , I was able to try the ```--force-dll-dependency-cache-update``` and that solved my problem. I opened up a PR to attempt to automatically detect such an error scenario where cached dependency can't be found, and retry with a forced dependency cache update. Ideally this would address the problem I faced for others (being confused on the compile error after my system update). Not sure if this is the best approach, or if any additional precautions need to be taken for cross-platform support, since my issue was on Windows. A link to the PR is here:
https://github.com/Nuitka/Nuitka/pull/206

I was also able to create a backup of my broken/affected cache before I used ```--force-dll-dependency-cache-update``` to solve the issue. I'll attach it below as well in case it is of any use to anyone in the future: 
[Nuitka.zip](https://github.com/Nuitka/Nuitka/files/2675589/Nuitka.zip)
